bmorph/evaluation/simple_river_network.py

```diff
@@ -217,14 +313,20 @@ def net_upstream_area(self, node: SegNode):
             returns the aggregate upstream_area
         """
         net_area = 0
-        if not node.end_marker:
+        if node and not node.end_marker:
             net_area = node.basin_area
-
-            for upstream_node in node.upstream:
-                net_area += self.net_upstream_area(upstream_node)
+            
+            if node.upstream:                                                             
+                for upstream_node in node.upstream:       
+                    try:      
+                        net_area += self.net_upstream_area(upstream_node)              
+                    except RecursionError as e:                  
+                        print(node, '\n', node.upstream,"\n", upstream_node,"\n", net_area)         
+                        raise RecursionError('no')
 
         return net_area
 
+    
```

Comment:
Make sure that within classes there is only 1 blank line between methods. Outside of classes there should be 2.